### PR TITLE
[`ruff`] Mark autofix for `RUF052` as always unsafe

### PR DESCRIPTION
**Pull request by AlexWaygood**
## Summary

#14819 improved the autofix offered by this rule for a bunch of standard-library classes that must be instantiated according to the pattern `T = TypeVar("T")`, where the string literal passed to the constructor _must_ match the name of the variable the call is being assigned to. However, there may be examples of this pattern in third-party libraries that we don't know about, so we can't assume that this fix is now safe.

TODO: needs a comment in `crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs` about why it's marked as unsafe

## Test Plan

`cargo test -p ruff_linter`

### CODE REVIEW

**Review by dylwil3 — APPROVED** [review]
Seems like the safest thing to do!

**Comment by AlexWaygood** [comment]
Thanks @dylwil3! I just pushed some docs to try to explain why it's marked as unsafe; could you take a quick second look?

File: crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs [hunk 1]
```diff
@@ -47,6 +52,16 @@ use crate::{checkers::ast::Checker, renamer::Renamer};
 /// would not shadow any other known variables already accessible from the scope
 /// in which the variable is defined.
 ///
+/// ## Fix safety
+/// This rule's fix is marked as unsafe.
+///
+/// As well as renaming the variable itself, the fix iterates through all references to the
+/// variable and adapts them to become references to the renamed variable. However, some renamings
```
Comment (dylwil3) [inline crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs:59]: How about:

```diff
- /// As well as renaming the variable itself, the fix iterates through all references to the
- /// variable and adapts them to become references to the renamed variable. However, some renamings
+ /// Ruff will attempt to rename the variable and all references to it. However, some renamings
```

**Review by dylwil3 — APPROVED** [review]
Looks good! Suggested possible rewording, but it's also fine as is

File: crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs (hunk 1 above)
Comment (AlexWaygood) [reply crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs:59]: Hmm... I like the improved concision... but I'm not sure about talking about "renaming references". I feel like, strictly speaking, the binding is renamed, but the references do not have a "name" as such (so they cannot be renamed), they only _point to_ (refer) to a name.

I'll push something that's more concise than what I've got currently, though ;)

Thanks!

**Review by Elno5 — COMMENTED** [review]
__.

**Pull request merged**